src/services/PersonInfoRequestProcessor.ts

```diff
@@ -0,0 +1,76 @@
+import { Metrics } from "@aws-lambda-powertools/metrics";
+import { Logger } from "@aws-lambda-powertools/logger";
+import  NodeRSA  from "node-rsa";
+import { F2fService } from "./F2fService";
+import { HttpCodesEnum } from "../models/enums/HttpCodesEnum";
+import { MessageCodes } from "../models/enums/MessageCodes";
+import { createDynamoDbClient } from "../utils/DynamoDBFactory";
+import { EnvironmentVariables } from "./EnvironmentVariables";
+import { personIdentityUtils } from "../utils/PersonIdentityUtils";
+import { Response } from "../utils/Response";
+import { ServicesEnum } from "../models/enums/ServicesEnum";
+import { APIGatewayProxyResult } from "aws-lambda";
+
+export class PersonInfoRequestProcessor {
+	private static instance: PersonInfoRequestProcessor;
+
+  	private readonly logger: Logger;
+
+  	private readonly metrics: Metrics;
+
+  	private readonly f2fService: F2fService;
+
+	private readonly publicKey: string;
+
+	private readonly environmentVariables: EnvironmentVariables;
+
+	constructor(logger: Logger, metrics: Metrics, publicKey: string) {
+		this.publicKey = publicKey;
+		this.environmentVariables = new EnvironmentVariables(logger, ServicesEnum.PERSON_INFO_SERVICE);
+		this.logger = logger;
+  		this.metrics = metrics;
+  		this.f2fService = F2fService.getInstance(this.environmentVariables.sessionTable(), this.logger, createDynamoDbClient());
+	}
+
+	static getInstance(logger: Logger, metrics: Metrics, publicKey: string): PersonInfoRequestProcessor {
+  	if (!PersonInfoRequestProcessor.instance) {
+  		PersonInfoRequestProcessor.instance = new PersonInfoRequestProcessor(logger, metrics, publicKey);
+  	}
+  	return PersonInfoRequestProcessor.instance;
+	}
+
+	async processRequest(sessionId: string): Promise<APIGatewayProxyResult> {
+  	const session = await this.f2fService.getSessionById(sessionId);
+		if (!session) {
+  		this.logger.error("No session found for session id", {
+  			messageCode: MessageCodes.SESSION_NOT_FOUND,
+  		});
+  		return Response(HttpCodesEnum.UNAUTHORIZED, `No session found with the session id: ${sessionId}`);
+  	}
+
+		this.logger.appendKeys({
+  		govuk_signin_journey_id: session?.clientSessionId,
+  	});
+
+  	const person = await this.f2fService.getPersonIdentityById(sessionId, this.environmentVariables.personIdentityTableName());
+		if (!person) {
+  		this.logger.error("No person found for session id", {
+  			messageCode: MessageCodes.PERSON_NOT_FOUND,
+  		});
+  		return Response(HttpCodesEnum.UNAUTHORIZED, `No person found with the session id: ${sessionId}`);
+  	}
+
+  	const address = personIdentityUtils.getStructuredPostalAddress(person.addresses[0], this.logger);
+  	const encryptedResponseValue = this.encryptResponse(address);
+
+  	return Response(HttpCodesEnum.OK, encryptedResponseValue);
+	}
+
+	encryptResponse(data: { address_line1: string; address_line2: string; town_city: string; postal_code: string }): string {
```

Comment:
we we use postalAddress instead of data here as this function is not accepting generic data.

Reply:
[PersonIdentityAddress](https://github.com/govuk-one-login/ipv-cri-f2f-api/blob/c1e8ac977b135f34921d4770c49054121abdbf41/src/models/PersonIdentityItem.ts#L10) contains fields that are not required for display in the FE. I made the choice to only encrypt required fields

Reply:
Fixed